packages/php-wasm/compile/build-assets/php_wasm.c

```diff
@@ -802,10 +815,38 @@ static void wasm_sapi_register_server_variables(zval *track_vars_array TSRMLS_DC
 	if (value != NULL) {
 		php_register_variable("SCRIPT_NAME", value, track_vars_array TSRMLS_CC);
 		php_register_variable("SCRIPT_FILENAME", value, track_vars_array TSRMLS_CC);
-		php_register_variable("PHP_SELF", value, track_vars_array TSRMLS_CC);
 		php_register_variable("REQUEST_URI", value, track_vars_array TSRMLS_CC);
 	}
 
+	int php_self_set = 0;
+	if (wasm_server_context->document_root != NULL && wasm_server_context->path_translated != NULL)
```

Comment:
[Proposal]: Maybe we should avoid null checks, in order to avoid accidental assignments?
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-if

Reply:
That works neatly in CPP but in C, `NULL` can have different values (`0` or `((void *)0)`). Perhaps we could use a linter rule instead?